lms/util/h_api.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+"""Utility functions for working with the h API."""
+
+from __future__ import unicode_literals
+
+import hashlib
+
+from lms.util.exceptions import MissingToolConsumerIntanceGUIDError
+from lms.util.exceptions import MissingUserIDError
+
+
+DISPLAY_NAME_MAX_LENGTH = 30
+"""The maximum length of an h display name."""
+
+
+USERNAME_MAX_LENGTH = 30
+"""The maximum length of an h username."""
+
+
+def generate_display_name(request_params):
+    """Return an h display_name from the given LTI launch request params."""
+    full_name = (request_params.get("lis_person_name_full") or "").strip()
+    given_name = (request_params.get("lis_person_name_given") or "").strip()
+    family_name = (request_params.get("lis_person_name_family") or "").strip()
+
+    if full_name:
+        display_name = full_name
+    else:
+        display_name = " ".join((given_name, family_name))
+
+    display_name = display_name.strip()
+
+    display_name = display_name or "Anonymous"
+
+    if len(display_name) > DISPLAY_NAME_MAX_LENGTH:
+        display_name = display_name[:DISPLAY_NAME_MAX_LENGTH - 1].rstrip() + "…"
+
+    return display_name
+
+
+def generate_username(request_params):
+    """
+    Return an h username generated from the given LTI launch request params.
+
+    :raises :py:exception:`lms.util.MissingToolConsumerIntanceGUIDError`:
+      if ``"tool_consumer_instance_guid"`` is missing from ``request_params``
+
+    :raises :py:exception:`lms.util.MissingUserIDError`:
+      if ``"user_id"`` is missing from ``request_params``
+    """
+    h = hashlib.sha1()  # pylint: disable=invalid-name
```

Comment:
Can we perhaps rename this variable? `h` threw me. It probably stands for `hash` but as we're working with the `h` API, it was confusing.

Reply:
Change to `hash_object` since `hash` is a builtin Python function